src/Security/Group.php

```diff
@@ -566,6 +547,21 @@ public function onBeforeWrite()
         if (!$this->Code && $this->Title != _t(__CLASS__ . '.NEWGROUP', "New Group")) {
             $this->setCode($this->Title);
         }
+
+        // Make sure the code for this group is unique.
+        // TODO in a future major version we should move this to validate() and throw a
```

Comment:
TODO's in comment tend to stick around for a while :-)

If we want to fix this in a future version, then we should move this code block to a separate private method and mark it as deprecated.  When preparing any future versions we'll look at all the deprecated methods and remove them, so this way we'll actually remember to swap the logic back to validate()

Reply:
Moved to a deprecated private method